bundle/src/main/java/com/adobe/acs/commons/dispatcher/impl/PermissionSensitiveCacheServlet.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * #%L
+ * ACS AEM Commons Bundle
+ * %%
+ * Copyright (C) 2017 - Adobe
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.adobe.acs.commons.dispatcher.impl;
+
+import org.apache.felix.scr.annotations.Component;
+import org.apache.felix.scr.annotations.Properties;
+import org.apache.felix.scr.annotations.Property;
+import org.apache.felix.scr.annotations.Service;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.resource.ResourceUtil;
+import org.apache.sling.api.servlets.SlingSafeMethodsServlet;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Component(
+        label = "ACS AEM Commons - Permission Sensitive Cache Servlet",
+        description = "Servlet that checks if the current sessions has access to a cached object",
+        metatype = true, immediate = true
+)
+@Properties({
+        @Property(
+                name = "sling.servlet.paths",
+                cardinality = Integer.MAX_VALUE,
+                label = "Sling Servlet Paths",
+                description = "Paths that this servlet will resolve to"
+        )
+})
+@Service
+public class PermissionSensitiveCacheServlet extends SlingSafeMethodsServlet {
+
+    private final Logger log = LoggerFactory.getLogger(PermissionSensitiveCacheServlet.class);
+
+
+    public void doHead(SlingHttpServletRequest request, SlingHttpServletResponse response) {
+        try{
+
+            //retrieve the requested URL
+            ResourceResolver resourceResolver = request.getResourceResolver();
+            String requestUri = request.getParameter( "uri" );
+
+            log.debug( "Checking access for URI {}", requestUri );
+
+            Resource requestedResource = resourceResolver.resolve( request, requestUri );
```

Comment:
@kaushalmall what happens when requestUri is null? Nothing bad right?

Reply:
@davidjgonzalez based on [0]

absPath - The absolute path to be resolved to a resource. `If this parameter is null, it is assumed to address the root of the resource tree`. If the path is relative it is assumed relative to the root, that is a slash is prepended to the path before resolving it.

Assuming this will be ok. I can put a null check too, if that's a better way of handling it. 

[0]https://sling.apache.org/apidocs/sling7/org/apache/sling/api/resource/ResourceResolver.html#resolve-javax.servlet.http.HttpServletRequest-java.lang.String-

Reply:
makes sense to me; easy enough check

Reply:
done. @davidjgonzalez